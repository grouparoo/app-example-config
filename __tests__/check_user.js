const { helper, relaxedSnapshot } = require("@grouparoo/spec-helper");

describe("check users", () => {
  // utility to start and stop the server within the test suite
  helper.grouparooTestServer({ truncate: true });

  test("mailchimp user", async () => {
    const { profile, snapshot } = await helper.getProfile({
      user_id: 22,
    });

    // You can do snapshot testing
    // `relaxedSnapshot` lets you skip time and UUID generated properties, but ensure everything else matches exactly
    expect(snapshot).toMatchSnapshot(relaxedSnapshot(snapshot));

    // Or you can test the properties of the snapshot directly
    expect(snapshot.groups.map((g) => g.name).sort()).toEqual(["all_emails"]);
    expect(snapshot.exports.map((g) => g.destinationName).sort()).toEqual([
      "newsletter",
    ]);

    const mailchimpData = snapshot.exports[0].newProfileProperties;
    const mailchimpTags = snapshot.exports[0].newGroups.sort();
    expect(mailchimpData).toEqual({
      FNAME: "Lisabeth",
      LNAME: "Auld",
      email_address: "lauldl@demo.com",
      language: "English",
    });
    expect(mailchimpTags).toEqual([]);
  });

  test("mailchimp tagged user", async () => {
    const { profile, snapshot } = await helper.getProfile({
      user_id: 88,
    });

    // You can do snapshot testing
    // `relaxedSnapshot` lets you skip time and UUID generated properties, but ensure everything else matches exactly
    expect(snapshot).toMatchSnapshot(relaxedSnapshot(snapshot));

    // Or you can test the properties of the snapshot directly
    expect(snapshot.groups.map((g) => g.name).sort()).toEqual([
      "all_emails",
      "high_value_spanish_speakers",
    ]);
    expect(snapshot.exports.map((g) => g.destinationName).sort()).toEqual([
      "newsletter",
    ]);

    const mailchimpData = snapshot.exports[0].newProfileProperties;
    const mailchimpTags = snapshot.exports[0].newGroups.sort();
    expect(mailchimpData).toEqual({
      FNAME: "Alena",
      LNAME: "Triggs",
      email_address: "atriggs2f@demo.com",
      language: "Spanish",
    });
    expect(mailchimpTags).toEqual(["High Value Spanish Speakers"]);
  });
});
