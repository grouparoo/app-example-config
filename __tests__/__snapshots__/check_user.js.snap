// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check users mailchimp tagged user 1`] = `
Object {
  "exports": Array [
    Object {
      "completedAt": null,
      "createdAt": null,
      "destination": null,
      "destinationName": "newsletter",
      "errorLevel": undefined,
      "errorMessage": undefined,
      "force": true,
      "hasChanges": true,
      "id": null,
      "mostRecent": false,
      "newGroups": Array [
        "High Value Spanish Speakers",
      ],
      "newProfileProperties": Object {
        "FNAME": "Alena",
        "LNAME": "Triggs",
        "email_address": "atriggs2f@demo.com",
        "language": "Spanish",
      },
      "oldGroups": Array [],
      "oldProfileProperties": Object {},
      "profileId": Any<String>,
      "startedAt": Any<Number>,
      "toDelete": false,
    },
  ],
  "groups": Array [
    Object {
      "calculatedAt": null,
      "createdAt": Any<Number>,
      "id": Any<String>,
      "locked": "config:code",
      "matchType": "all",
      "name": "all_emails",
      "nextCalculatedAt": Any<Number>,
      "profilesCount": 2,
      "rules": Array [
        Object {
          "key": "email",
          "match": "null",
          "operation": Object {
            "description": "is not equal to",
            "op": "ne",
          },
          "relativeMatchDirection": null,
          "relativeMatchNumber": null,
          "relativeMatchUnit": null,
          "topLevel": false,
          "type": "email",
        },
      ],
      "state": "updating",
      "type": "calculated",
      "updatedAt": Any<Number>,
    },
    Object {
      "calculatedAt": null,
      "createdAt": Any<Number>,
      "id": Any<String>,
      "locked": "config:code",
      "matchType": "all",
      "name": "high_value_spanish_speakers",
      "nextCalculatedAt": Any<Number>,
      "profilesCount": 1,
      "rules": Array [
        Object {
          "key": "ltv",
          "match": "100",
          "operation": Object {
            "description": "is greater than",
            "op": "gt",
          },
          "relativeMatchDirection": null,
          "relativeMatchNumber": null,
          "relativeMatchUnit": null,
          "topLevel": false,
          "type": "float",
        },
        Object {
          "key": "language",
          "match": "Spanish",
          "operation": Object {
            "description": "is equal to",
            "op": "eq",
          },
          "relativeMatchDirection": null,
          "relativeMatchNumber": null,
          "relativeMatchUnit": null,
          "topLevel": false,
          "type": "string",
        },
      ],
      "state": "ready",
      "type": "calculated",
      "updatedAt": Any<Number>,
    },
  ],
  "properties": Object {
    "email": Object {
      "confirmedAt": Any<Date>,
      "createdAt": Any<Date>,
      "id": Any<String>,
      "identifying": false,
      "isArray": false,
      "startedAt": null,
      "state": "ready",
      "stateChangedAt": Any<Date>,
      "type": "email",
      "unique": true,
      "updatedAt": Any<Date>,
      "valueChangedAt": Any<Date>,
      "values": Array [
        "atriggs2f@demo.com",
      ],
    },
    "first_name": Object {
      "confirmedAt": Any<Date>,
      "createdAt": Any<Date>,
      "id": Any<String>,
      "identifying": false,
      "isArray": false,
      "startedAt": null,
      "state": "ready",
      "stateChangedAt": Any<Date>,
      "type": "string",
      "unique": false,
      "updatedAt": Any<Date>,
      "valueChangedAt": Any<Date>,
      "values": Array [
        "Alena",
      ],
    },
    "language": Object {
      "confirmedAt": Any<Date>,
      "createdAt": Any<Date>,
      "id": Any<String>,
      "identifying": false,
      "isArray": false,
      "startedAt": null,
      "state": "ready",
      "stateChangedAt": Any<Date>,
      "type": "string",
      "unique": false,
      "updatedAt": Any<Date>,
      "valueChangedAt": Any<Date>,
      "values": Array [
        "Spanish",
      ],
    },
    "last_name": Object {
      "confirmedAt": Any<Date>,
      "createdAt": Any<Date>,
      "id": Any<String>,
      "identifying": false,
      "isArray": false,
      "startedAt": null,
      "state": "ready",
      "stateChangedAt": Any<Date>,
      "type": "string",
      "unique": false,
      "updatedAt": Any<Date>,
      "valueChangedAt": Any<Date>,
      "values": Array [
        "Triggs",
      ],
    },
    "ltv": Object {
      "confirmedAt": Any<Date>,
      "createdAt": Any<Date>,
      "id": Any<String>,
      "identifying": false,
      "isArray": false,
      "startedAt": null,
      "state": "ready",
      "stateChangedAt": Any<Date>,
      "type": "float",
      "unique": false,
      "updatedAt": Any<Date>,
      "valueChangedAt": Any<Date>,
      "values": Array [
        123.67000000000002,
      ],
    },
    "purchase_count": Object {
      "confirmedAt": Any<Date>,
      "createdAt": Any<Date>,
      "id": Any<String>,
      "identifying": false,
      "isArray": false,
      "startedAt": null,
      "state": "ready",
      "stateChangedAt": Any<Date>,
      "type": "integer",
      "unique": false,
      "updatedAt": Any<Date>,
      "valueChangedAt": Any<Date>,
      "values": Array [
        3,
      ],
    },
    "user_id": Object {
      "confirmedAt": Any<Date>,
      "createdAt": Any<Date>,
      "id": Any<String>,
      "identifying": true,
      "isArray": false,
      "startedAt": null,
      "state": "ready",
      "stateChangedAt": Any<Date>,
      "type": "integer",
      "unique": true,
      "updatedAt": Any<Date>,
      "valueChangedAt": Any<Date>,
      "values": Array [
        88,
      ],
    },
  },
}
`;

exports[`check users mailchimp user 1`] = `
Object {
  "exports": Array [
    Object {
      "completedAt": null,
      "createdAt": null,
      "destination": null,
      "destinationName": "newsletter",
      "errorLevel": undefined,
      "errorMessage": undefined,
      "force": true,
      "hasChanges": true,
      "id": null,
      "mostRecent": false,
      "newGroups": Array [],
      "newProfileProperties": Object {
        "FNAME": "Lisabeth",
        "LNAME": "Auld",
        "email_address": "lauldl@demo.com",
        "language": "English",
      },
      "oldGroups": Array [],
      "oldProfileProperties": Object {},
      "profileId": Any<String>,
      "startedAt": Any<Number>,
      "toDelete": false,
    },
  ],
  "groups": Array [
    Object {
      "calculatedAt": null,
      "createdAt": Any<Number>,
      "id": Any<String>,
      "locked": "config:code",
      "matchType": "all",
      "name": "all_emails",
      "nextCalculatedAt": Any<Number>,
      "profilesCount": 1,
      "rules": Array [
        Object {
          "key": "email",
          "match": "null",
          "operation": Object {
            "description": "is not equal to",
            "op": "ne",
          },
          "relativeMatchDirection": null,
          "relativeMatchNumber": null,
          "relativeMatchUnit": null,
          "topLevel": false,
          "type": "email",
        },
      ],
      "state": "updating",
      "type": "calculated",
      "updatedAt": Any<Number>,
    },
  ],
  "properties": Object {
    "email": Object {
      "confirmedAt": Any<Date>,
      "createdAt": Any<Date>,
      "id": Any<String>,
      "identifying": false,
      "isArray": false,
      "startedAt": null,
      "state": "ready",
      "stateChangedAt": Any<Date>,
      "type": "email",
      "unique": true,
      "updatedAt": Any<Date>,
      "valueChangedAt": Any<Date>,
      "values": Array [
        "lauldl@demo.com",
      ],
    },
    "first_name": Object {
      "confirmedAt": Any<Date>,
      "createdAt": Any<Date>,
      "id": Any<String>,
      "identifying": false,
      "isArray": false,
      "startedAt": null,
      "state": "ready",
      "stateChangedAt": Any<Date>,
      "type": "string",
      "unique": false,
      "updatedAt": Any<Date>,
      "valueChangedAt": Any<Date>,
      "values": Array [
        "Lisabeth",
      ],
    },
    "language": Object {
      "confirmedAt": Any<Date>,
      "createdAt": Any<Date>,
      "id": Any<String>,
      "identifying": false,
      "isArray": false,
      "startedAt": null,
      "state": "ready",
      "stateChangedAt": Any<Date>,
      "type": "string",
      "unique": false,
      "updatedAt": Any<Date>,
      "valueChangedAt": Any<Date>,
      "values": Array [
        "English",
      ],
    },
    "last_name": Object {
      "confirmedAt": Any<Date>,
      "createdAt": Any<Date>,
      "id": Any<String>,
      "identifying": false,
      "isArray": false,
      "startedAt": null,
      "state": "ready",
      "stateChangedAt": Any<Date>,
      "type": "string",
      "unique": false,
      "updatedAt": Any<Date>,
      "valueChangedAt": Any<Date>,
      "values": Array [
        "Auld",
      ],
    },
    "ltv": Object {
      "confirmedAt": Any<Date>,
      "createdAt": Any<Date>,
      "id": Any<String>,
      "identifying": false,
      "isArray": false,
      "startedAt": null,
      "state": "ready",
      "stateChangedAt": Any<Date>,
      "type": "float",
      "unique": false,
      "updatedAt": Any<Date>,
      "valueChangedAt": Any<Date>,
      "values": Array [
        162.78,
      ],
    },
    "purchase_count": Object {
      "confirmedAt": Any<Date>,
      "createdAt": Any<Date>,
      "id": Any<String>,
      "identifying": false,
      "isArray": false,
      "startedAt": null,
      "state": "ready",
      "stateChangedAt": Any<Date>,
      "type": "integer",
      "unique": false,
      "updatedAt": Any<Date>,
      "valueChangedAt": Any<Date>,
      "values": Array [
        3,
      ],
    },
    "user_id": Object {
      "confirmedAt": Any<Date>,
      "createdAt": Any<Date>,
      "id": Any<String>,
      "identifying": true,
      "isArray": false,
      "startedAt": null,
      "state": "ready",
      "stateChangedAt": Any<Date>,
      "type": "integer",
      "unique": true,
      "updatedAt": Any<Date>,
      "valueChangedAt": Any<Date>,
      "values": Array [
        22,
      ],
    },
  },
}
`;
